lib/spack/spack/relocate.py

```diff
@@ -240,44 +229,67 @@ def needs_text_relocation(filetype):
     """
     Check whether the given filetype is text that may need relocation.
     """
-    return ("text" in filetype)
+    return ("ASCII text" in filetype)
 
 
-def relocate_binary(path_name, old_dir, new_dir):
+def relocate_binary(path_names, old_dir, new_dir):
     """
     Change old_dir to new_dir in RPATHs of elf or mach-o file path_name
     """
     if platform.system() == 'Darwin':
-        modify_macho_object(path_name, old_dir, new_dir, relative=False)
+        for path_name in path_names:
+            rpaths, deps, idpath = macho_get_paths(path_name)
+            nid = None
+            nrpaths = []
+            ndeps = []
+            nrpaths, ndeps, nid = macho_replace_paths(old_dir, new_dir,
+                                                      rpaths, deps, idpath)
+            modify_macho_object(path_name,
+                                rpaths, deps, idpath,
+                                nrpaths, ndeps, nid)
     elif platform.system() == 'Linux':
-        orig_rpaths = get_existing_elf_rpaths(path_name)
-        new_rpaths = substitute_rpath(orig_rpaths, old_dir, new_dir)
-        modify_elf_object(path_name, orig_rpaths, new_rpaths)
+        for path_name in path_names:
+            orig_rpaths = get_existing_elf_rpaths(path_name)
+            new_rpaths = substitute_rpath(orig_rpaths, old_dir, new_dir)
+            modify_elf_object(path_name, orig_rpaths, new_rpaths)
     else:
         tty.die("Relocation not implemented for %s" % platform.system())
 
 
-def make_binary_relative(cur_path_name, orig_path_name, old_dir):
+def make_binary_relative(cur_path_names, orig_path_names, old_dir):
     """
     Make RPATHs relative to old_dir in given elf or mach-o file path_name
     """
     if platform.system() == 'Darwin':
-        new_dir = ''
-        modify_macho_object(cur_path_name, orig_path_name, old_dir, new_dir,
-                            relative=True)
+        for cur_path_name, orig_path_name in zip(cur_path_names,
+                                                 orig_path_names):
+            rpaths, deps, idpath = macho_get_paths(cur_path_name)
+            nid = None
```

Comment:
This and the one above were my attempt to fix the real bug which was not setting id=None in macho_get_paths.

Reply:
What I meant was that when you do `nrpaths, ndeps, nid = macho_make_paths_relative(...` below it will overwrite the value you are setting here, so I don't see a need to set `nid` here.

Reply:
Goes back to the bug in macho_get_paths. I set idpath="" and when it was returned and passed to macho_make_paths_relative it caused and error in the  if idpath: statement. Now that I know the source of the bug I will remove these assignments.